packages/public/store/src/history/reducer.ts

```diff
@@ -104,6 +104,16 @@ export const hasPendingChanges: Selector<boolean> = createSelector(
   (state) => getPendingChanges()(state) !== 0
 )
 
+/** @public */
+export const hasUndoActions: Selector<boolean> = createSelector(
+  (state) => getHistory()(state as InternalScopedState).undoStack.length > 0
```

Comment:
Instead of the type assertion, it's probably cleaner to expose the state of the history, i.e. https://github.com/edtr-io/edtr-io/blob/fd6290d7ab4b7e493f35e7219f10bbb6bf912868/packages/public/store/src/types.ts#L25
Something like:
```tsc
{
  history: {
    undoStack: unknown[],
    redoStack: unknown[],
    pendingChanges: number
  }
}
```
still hides most of the implementation details while exposing the stuff our public selectors need.